lib/jira/install.js

```diff
@@ -1,28 +1,43 @@
 const { Installation } = require('../models')
 const getAxiosInstance = require('../jira/client/axios')
+const jwt = require('./util/jwt')
 
 module.exports = async (req, res) => {
   req.log('Received installation payload.')
   const { baseUrl: host, clientKey, sharedSecret } = req.body
-  res.sendStatus(204)
+  let successfulRes = false;
+
+  try {
+    const hostData = await Installation.getForClientKey(clientKey)
+    if (jwt.hasValidJwt(hostData.sharedSecret, host, req, res)) {
+      res.sendStatus(204)
+      successfulRes = true
+    }
+  } catch (err) {
+    // Client key has not been seen before. Safe to install.
+    res.sendStatus(204)
+    successfulRes = true
+  }
   res.on('finish', async () => {
-    const instance = getAxiosInstance(null, {}, host, sharedSecret)
-    try {
-      // Validate the clientKey and the sharedSecret
-      // We have to do this after responding with 204 because
-      // otherwise we get an error "The add-on is not enabled"
-      // no matter what we send
-      const result = await instance.get(`/rest/devinfo/0.10/existsByProperties?fakeProperty=1`)
-      if (result.status === 200) {
-        req.log('App installed on Jira. Adding secrets.')
-        await Installation.install({ host, clientKey, sharedSecret })
-      } else {
-        req.log('App not installed. Verification failed.')
-      }
-    } catch (err) {
-      if (err.response.status === 401) {
-        req.log('Jira does not recognize this installation. Deleting it.')
-        await Installation.uninstall({ clientKey })
+    if (successfulRes) {
```

Comment:
I believe I could also use `res.statusCode === 204`, but I couldnt start this up to try it out :/

Reply:
In order to simplify the code indentation and the diff, can you do `if (!successfulRes) return`?

Reply:
Done :)